test/unittests/utils/test_utils_inputs.py

```diff
@@ -0,0 +1,591 @@
+
+    information = {
+        "pattern": "*space-MNI152NLin2009cSym_res-1x1x1_T1w.nii.gz",
+        "description": "T1w image registered in MNI152NLin2009cSym space using t1-linear pipeline",
+        "needed_pipeline": "t1-linear",
+    }
+
+    with pytest.raises(
+        ValueError,
+        match="Subjects and sessions must have the same length.",
+    ):
+        clinica_file_reader(
+            ["sub-01"],
+            ["ses-M00", "ses-M06"],
+            tmp_path,
+            information,
+            raise_exception=True,
+            n_procs=1,
+        )
+
+    assert clinica_file_reader(
+        [], [], tmp_path, information, raise_exception=True, n_procs=1
+    ) == ([], "")
+
+    results, error_msg = clinica_file_reader(
+        ["sub-01"], ["ses-M00"], tmp_path, information, raise_exception=True, n_procs=1
+    )
+    assert len(results) == 1
+    expected_error_msg = (
+        "Clinica encountered 0 problem(s) while getting T1w image registered "
+        "in MNI152NLin2009cSym space using t1-linear pipeline:\n"
+        "Please note that the following clinica pipeline(s) must have run to "
+        "obtain these files: t1-linear\n"
+    )
+    assert error_msg == expected_error_msg
+
+    results, error_msg = clinica_file_reader(
+        ["sub-01", "sub-02", "sub-02", "sub-06"],
+        ["ses-M00", "ses-M00", "ses-M06", "ses-M00"],
+        tmp_path,
+        information,
+        raise_exception=True,
+        n_procs=4,
+    )
+    assert len(results) == 4
+    assert error_msg == expected_error_msg
+
+    (
+        tmp_path
+        / "subjects"
+        / "sub-01"
+        / "ses-M00"
+        / "t1_linear"
+        / "sub-01_ses-M00_foo-bar_T1w_space-MNI152NLin2009cSym_res-1x1x1_T1w.nii.gz"
+    ).mkdir()
+    results, error_msg = clinica_file_reader(
+        ["sub-01"], ["ses-M00"], tmp_path, information, raise_exception=False, n_procs=1
+    )
+    assert len(results) == 0
+    expected_msg = (
+        "Clinica encountered 1 problem(s) while getting T1w image registered "
+        "in MNI152NLin2009cSym space using t1-linear pipeline:\n"
+        "Please note that the following clinica pipeline(s) must have run to "
+        "obtain these files: t1-linear\n"
+        "\t*  (sub-01 | ses-M00): More than 1 file found:\n"
+    )
+    assert expected_msg in error_msg
+    with pytest.raises(
+        ClinicaCAPSError,
+    ):
+        clinica_file_reader(
+            ["sub-01"],
+            ["ses-M00"],
+            tmp_path,
+            information,
+            raise_exception=True,
+            n_procs=1,
+        )
+
+
+def test_clinica_list_of_files_reader(tmp_path):
+    from clinica.utils.exceptions import ClinicaBIDSError
+    from clinica.utils.inputs import clinica_list_of_files_reader
+
+    config = {
+        "sub-01": ["ses-M00"],
+        "sub-02": ["ses-M00", "ses-M06"],
+        "sub-06": ["ses-M00"],
+    }
+
+    build_bids_directory(tmp_path, config)
+
+    informations = [
```

Comment:
Typo: In word 'informations'

PyCharm's spell checker suggests renaming to 'information'